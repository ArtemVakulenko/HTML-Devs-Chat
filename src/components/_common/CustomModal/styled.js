import styled from 'styled-components';

export const StCustomModal = styled.div`
    width: 25%;
    height: 90%;
    position: fixed;
    display: flex;
    flex-direction: column;
    align-content: center;
    background-color: ${({ colors, theme }) => colors[theme].secondary};
    border: 2px solid ${({ colors, theme }) => colors[theme].modal};
    border-radius: 10px;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
`;

export const StModalWrapper = styled.div`
    top: 0;
    left: 0;
    position: absolute;
    background: ${({ colors, theme }) => colors[theme].modalWrapper};
    width: 100%;
    height: 100%;
`;

export const StModalHeader = styled.header`
    padding: 5px;
    width: 100%;
    display: flex;
`;
export const StModalContent = styled.div`
    width:  100%;
    height: calc(100% - 80px);
    padding: 10px;
    display: flex;
    flex-direction: column;
`;
