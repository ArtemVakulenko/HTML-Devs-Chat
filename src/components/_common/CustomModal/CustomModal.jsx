import React from 'react';
import PropTypes from 'prop-types';
import { icons } from 'constants';
import { StCustomModal, StModalHeader, StModalWrapper, StModalContent } from './styled';
import { useTheme } from '../../hooks';
import Button from '../Button';
import Title from '../Title';

const CustomModal = ({ title, children, onClose }) => {
  const { colors, theme } = useTheme();
  const onCloseModal = (e) => {
    if (e.target.id === 'modalWrapper') {
      onClose();
    }
  };
  return (
    <StModalWrapper colors={colors} theme={theme} id='modalWrapper' onClick={onCloseModal}>
      <StCustomModal colors={colors} theme={theme}>
        <StModalHeader>
          <Title title={title} />
          <Button onClick={onClose} icon={icons.closeModalDark} />
        </StModalHeader>
        <StModalContent>
          {children}
        </StModalContent>
      </StCustomModal>
    </StModalWrapper>
  );
};

export default CustomModal;

CustomModal.propTypes = {
  title: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired,
};
