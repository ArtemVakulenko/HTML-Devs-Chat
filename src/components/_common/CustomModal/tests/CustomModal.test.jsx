import React from 'react';
import { shallow, mount } from 'enzyme';
import CustomModal from '../CustomModal';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);
jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));

describe('CustomModal', () => {
  const props = {
    title: 'My account',
    onClose: jest.fn(),
    children: <div />,
  };
  it('Should match snapshot', () => {
    const component = shallow(<CustomModal {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StModalWrapper ', () => {
    const component = mount(<CustomModal {...props} />);
    expect(component.find('styled__StModalWrapper')).toHaveLength(1);
  });
  it('Should have Title', () => {
    const component = mount(<CustomModal {...props} />);
    expect(component.find('Title')).toHaveLength(1);
  });
  it('Should have button', () => {
    const component = mount(<CustomModal {...props} />);
    expect(component.find('Button')).toHaveLength(1);
  });
  it('Should have StModalHeader ', () => {
    const component = mount(<CustomModal {...props} />);
    expect(component.find('styled__StModalHeader')).toHaveLength(1);
  });
  it('Should have StModalContent ', () => {
    const component = mount(<CustomModal {...props} />);
    expect(component.find('styled__StModalContent')).toHaveLength(1);
  });
  it('Should close modal on click on the StModalWrapper', () => {
    const component = mount(<CustomModal {...props} />);
    component.find('styled__StModalWrapper').simulate('click');
    expect(component.props().onClose).toHaveBeenCalled();
  });
});
