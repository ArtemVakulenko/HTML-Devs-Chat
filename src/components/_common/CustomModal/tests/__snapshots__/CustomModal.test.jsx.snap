// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal Should match snapshot 1`] = `"<div id=\\"modalWrapper\\" class=\\"styled__StModalWrapper-hYaZKU XesCm\\"><div class=\\"styled__StCustomModal-iPKCGo lfyqOZ\\"><header class=\\"styled__StModalHeader-iEAzNg fHeyVw\\"><h1 class=\\"styled__AppTitle-bMNsHC jWDbjD\\"></h1><button class=\\"styled__StIcon-chPeWD giwZPn\\"></button></header><div class=\\"styled__StModalContent-jgQuhi ilBLET\\"><div></div></div></div></div>"`;
